### PR TITLE
refactor: Move http delete parsing code into router along side other http code

### PR DESCRIPTION
**Pull request by alamb**
~Draft as it builds on https://github.com/influxdata/influxdb_iox/pull/5840~

# Rationale
Implement a TODO from long ago and move HTTP specific code next to the http handlers

### CODE REVIEW

File: router/Cargo.toml [hunk 1]
```diff
@@ -26,9 +26,11 @@ parking_lot = "0.12"
 predicate = { path = "../predicate" }
 schema = { version = "0.1.0", path = "../schema" }
 serde = "1.0"
+serde_json = "1.0.86"
 serde_urlencoded = "0.7"
 service_grpc_schema = { path = "../service_grpc_schema" }
 service_grpc_object_store = { path = "../service_grpc_object_store" }
+snafu = "0.7"
```
Comment (alamb) [inline router/Cargo.toml:33]: I think this is the most potentially controversial change in this PR

The rest of the router code uses a different error handling pattern / library (`thiserror`) than other parts of the system (`snafu`). So when I ported the code from elsewhere now router uses snafu

If this is a problem, I can rework the code to use thiserror, though I would rather not as I don't think the value of doing so is very high compared to the benefit.

File: predicate/src/delete_predicate.rs [hunk 2]
```diff
@@ -35,30 +33,6 @@ pub enum Error {
     /// Predicate include non supported expression
     #[snafu(display("Delete predicate must be conjunctive expressions of binary 'column_name = literal' or 'column_name != literal': ({})", value))]
     NotSupportPredicate { value: String },
-
```
Comment (alamb) [inline predicate/src/delete_predicate.rs:38]: This code and test is just all mover

**Review by domodwyer — APPROVED** [review]
We could also just gift it to the history of git, given recent developments.

**Comment by alamb** [comment]
> We could also just gift it to the history of git, given recent developments.

Yeah, if I had to do anything more than move it around I will do so... Sometimes it is so hard to https://www.youtube.com/watch?v=FnpJBkAMk44

**Pull request merged**